server/core/db/boltdb.go

```diff
@@ -106,23 +114,80 @@ func (b *BoltDB) TryLock(newLock models.ProjectLock) (bool, models.ProjectLock,
 		if err := json.Unmarshal(currLockSerialized, &currLock); err != nil {
 			return errors.Wrap(err, "failed to deserialize current lock")
 		}
+
+		// checking if current lock is with the same PR
+		if currLock.Pull.Num == newLock.Pull.Num {
+			lockAcquired = true
+			return nil
+		}
+
 		lockAcquired = false
+
+		// Enqueuing
+		queueBucket := tx.Bucket(b.queueBucketName)
+		currQueueSerialized := queueBucket.Get([]byte(key))
+		// Queue doesn't exist, create one
+		if currQueueSerialized == nil {
+			newQueue := models.ProjectLockQueue{newLock}
+			newQueueSerialized, err := json.Marshal(newQueue)
+			if err != nil {
+				return errors.Wrap(err, "serializing")
+			}
+			if err := queueBucket.Put([]byte(key), newQueueSerialized); err != nil {
+				return err
```

Comment:
Why not use `errors.Wrap()` here as well?

Reply:
I'd pick one and go with that. The code goes back and forth but this pr should pick one to be consistent if possible

Reply:
98b5851